### ISSUE TITLE
Parser treats components named after HTML void elements as self-closing

### ISSUE THREAD

**Issue by alasano** [pheuter/svelte-check-rs#113]
## Summary

The parser incorrectly treats PascalCase Svelte components as void (self-closing) elements when their lowercased name matches an HTML void element. This causes false "mismatched closing tag" errors and cascading parse failures.

## Affected Component Names

Any component whose lowercased name collides with an HTML void element: `Input`, `Img`, `Link`, `Meta`, `Source`, `Area`, `Base`, `Br`, `Col`, `Embed`, `Hr`, `Track`, `Wbr`.

## Reproduction

```svelte
<Form>
  <Input type="radio" name="group" value="a" bind:group={selected}>
    {#snippet label()}
      Option A
    {/snippet}
  </Input>
</Form>
```

**Expected:** Parses without errors.

**Actual:** Reports `mismatched closing tag: expected </Form>, found </Input>` and cascading errors.

## Root Cause

In `crates/svelte-parser/src/parser.rs`, `is_void_element()` performs a case-insensitive lookup, so `Input` matches `input` in the void elements list. The check does not account for the fact that PascalCase names are components, not HTML elements.

## Impact

A single `<Input>...</Input>` with children produces ~5 cascading parse errors (mismatched tags for every subsequent sibling and ancestor). In a real codebase, this can add dozens of false errors on top of legitimate diagnostics.
